awx/main/middleware.py

```diff
@@ -162,3 +164,12 @@ def process_request(self, request):
         if request.path_info != new_path:
             request.path = request.path.replace(request.path_info, new_path)
             request.path_info = new_path
+
+
+class MigrationRanCheckMiddleware(object):
+
+    def process_request(self, request):
+        executor = MigrationExecutor(connection)
+        plan = executor.migration_plan(executor.loader.graph.leaf_nodes())
```

Comment:
Hah, this is neat. Wouldn't of thought of that.

Reply:
Is this expensive?

Reply:
Not that I've seen, I ran through it a couple of times. Building the initial list of "What does this need to do" seems pretty small.

I'd totally rather have this be something that happens as part of the installer that swaps out the nginx config when the migrations are done but we've gone down the path of separating task execution from web service.